### PR TITLE
spacedrive: init at 0.1.4

### LINKED ISSUE #170254: spacedrive: Packaging request
**Project description**
Cross platform file manager supporting cloud providers

**Metadata**

* homepage URL: https://www.spacedrive.app/
* source URL: https://github.com/spacedriveapp/spacedrive
* license: GPL-3.0
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by MikaelFangel**
## Description of changes

Fixes #170254

This adds the spacedrive application to nixpkgs.

@heisfer: I have merged my NUR derivation with yours. Thank you! Do you want to co-maintain this package?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sp/spacedrive/package.nix
Comment (MikaelFangel) [reply pkgs/by-name/sp/spacedrive/package.nix]: @pbsds - We need the variable name as it is the one deciding the name for the binary, so I've kept it, so the binary won't be named `spacedrive-0.1.4`.

Comment (MikaelFangel) [reply pkgs/by-name/sp/spacedrive/package.nix]: The rule about the name variable can be found here:
[https://github.com/NixOS/nixpkgs/blob/5689f242a1532ee4281e64fa98170db18ce6d234/doc/build-helpers/images/appimagetools.section.md](https://github.com/NixOS/nixpkgs/blob/5689f242a1532ee4281e64fa98170db18ce6d234/doc/build-helpers/images/appimagetools.section.md)

**Comment by heisfer** [comment]
Yeah, you could add me.
"name" variable isn't really used so don't really see the point of being there.

**Comment by MikaelFangel** [comment]
Should we just move the package to avoid having the name variable, then? The name variable is used to name the binary. If it's not present, and we don't manually move the binary, the name will become spacedrive-0.1.4.

[https://github.com/NixOS/nixpkgs/blob/5689f242a1532ee4281e64fa98170db18ce6d234/doc/build-helpers/images/appimagetools.section.md](https://github.com/NixOS/nixpkgs/blob/5689f242a1532ee4281e64fa98170db18ce6d234/doc/build-helpers/images/appimagetools.section.md)

**Comment by heisfer** [comment]
technically it's a file-explorer so it could be moved to pkgs/applications/file-managers but I'm no expert on this field so @pbsds could give few tips.

**Comment by MikaelFangel** [comment]
The derivation should stay under the `pkgs/by-name/so/some-example/package.nix` as it is the preferred way of adding new packages to nixpkgs.

**Comment by heisfer** [comment]
then it's fine i think

**Comment by MikaelFangel** [comment]
To add you as a maintainer you need to appear in the maintainers list (: You should do that by adding your self and referring to this PR.

**Review by pbsds — COMMENTED** [review]
Wow, forgot that bit about `appimageTools`. Not the best design IMO. The suggested fix is a pattern i've seen elsewhere.

File: pkgs/by-name/sp/spacedrive/package.nix
Comment (pbsds) [reply pkgs/by-name/sp/spacedrive/package.nix]: doh, not the best design

**Comment by MikaelFangel** [comment]
@pbsds — I've seen that pattern too, but I think we should get entirely rid of the name variable then. I've committed your suggestion together with the change where I remove the name variable.

File: pkgs/by-name/sp/spacedrive/package.nix
Comment (pbsds) [reply pkgs/by-name/sp/spacedrive/package.nix]: fun fact: `name` defaults to `${pname}-${version}`

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 270121` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

LGTM, ping me for merge if heisfer doesn't respond :)

File: pkgs/by-name/sp/spacedrive/package.nix
Comment (MikaelFangel) [reply pkgs/by-name/sp/spacedrive/package.nix]: Appimage tools should really be updated so that they are more similar to other nix tools 😅

**Pull request merged**
